omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_12/OrderGroupController1_12Test.java

```diff
@@ -0,0 +1,176 @@
+			"      \"careSetting\": \"6f0c9a92-6f24-11e3-af88-005056821db0\",\n" + 
+			"      \"autoExpireDate\": \"2012-09-21\",\n" + 
+			"      \"orderer\": \"c2299800-cca9-11e0-9572-0800200c9a66\",\n" + 
+			"      \"previousOrder\": \"\",\n" + 
+			"      \"urgency\": \"ROUTINE\",\n" + 
+			"      \"orderReason\": \"\",\n" + 
+			"      \"orderReasonNonCoded\": \"string\",\n" + 
+			"      \"instructions\": \"string\",\n" + 
+			"      \"commentToFulfiller\": \"string\"\n" + 
+			"    }\n" + 
+			"  ],\n" + 
+			"  \"orderSet\": \"2d3fb1d0-ae06-22e3-a5e2-0140211c1001\"\n" + 
+			"}";
+	
+	String json_order = " { \"orders\": [\n" + 
+			"    {\n" + 
+			"      \"encounter\": \"33d70956-b359-452a-b3da-b69c8ab459ce\",\n" + 
+			"      \"orderType\": \"2d3fb1d0-ae06-22e3-a5e2-0140211c9a66\",\n" + 
+			"      \"type\": \"order\",\n" + 
+			"      \"action\": \"NEW\",\n" + 
+			"      \"accessionNumber\": \"string\",\n" + 
+			"      \"dateActivated\": \"2012-09-11\",\n" + 
+			"      \"scheduledDate\": \"2012-09-11\",\n" + 
+			"      \"patient\": \"19887889-7110-46a7-a9f0-ebf1eb4bbd48\",\n" + 
+			"      \"concept\": \"985ff1a2-c2ef-49fd-836f-8a1d936d9ef9\",\n" + 
+			"      \"careSetting\": \"6f0c9a92-6f24-11e3-af88-005056821db0\",\n" + 
+			"      \"autoExpireDate\": \"2012-09-21\",\n" + 
+			"      \"orderer\": \"c2299800-cca9-11e0-9572-0800200c9a66\",\n" + 
+			"      \"previousOrder\": \"\",\n" + 
+			"      \"urgency\": \"ROUTINE\",\n" + 
+			"      \"orderReason\": \"\",\n" + 
+			"      \"orderReasonNonCoded\": \"for Test\",\n" + 
+			"      \"instructions\": \"string\",\n" + 
+			"      \"commentToFulfiller\": \"string\"\n" + 
+			"    }\n" +
+			"  ]\n" +
+			"}";
+	@Before
+	public void init() throws Exception {
+		orderService = Context.getOrderService();
+		executeDataSet(RestTestConstants1_12.ORDER_GROUP_TEST_DATA_SET);
+	}
+	@Override
+	public String getURI() {
+
+		return "ordergroup";
+	}
+
+	@Override
+	public String getUuid() {
+		return RestTestConstants1_12.ORDER_GROUP_UUID;
+	}
+
+	@Override
+	public long getAllCount()  {
+		return 0;
+	}
+
+	@Test
+	public void shouldCreateNewOrderGroup() throws Exception{
+		   //Creating a new OrderGroup with  orders
+		MockHttpServletRequest req = newPostRequest(getURI(), json);
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertEquals(patient_uuid, Util.getByPath(result, "patient/uuid"));
+		Assert.assertEquals(encounter_uuid,Util.getByPath(result, "encounter/uuid"));
+		Assert.assertEquals(orderSet_uuid,Util.getByPath(result, "orderSet/uuid"));
+		Assert.assertEquals(display, Util.getByPath(result, "display"));
+				
+		String uuid =  (String) PropertyUtils.getProperty(result, "uuid");
+		Assert.assertEquals(2, orderService.getOrderGroupByUuid(uuid).getOrders().size());			
+	}
+	
+	@Test
+	public void shouldGetOrderGroupByUuid() throws Exception {
+		MockHttpServletRequest post_req = newPostRequest(getURI(), json);
```

Comment:
If you are testing a Get, then why are you doing a POST?

Reply:
well i was just adding a few orders to the order Group, but let me adjust it

Reply:
i  have removed the Post from the test for fetching